test/TempleLineOfCredit.t.sol

```diff
@@ -0,0 +1,292 @@
+        assertEq(debtBalanceBefore + repayAmount, tlc.debtBalance());
+        assertEq(block.timestamp, aliceCreatedAt);
+    }
+
+    function testWithdrawExceedCollateralAmount() external {
+        uint256 borrowAmount = uint(60_000e18);
+        uint256 collateralAmount = uint(100_000e18);
+        uint256 withdrawalAmount = uint(100_001e18);
+        _borrow(alice, collateralAmount, borrowAmount);
+         vm.expectRevert(abi.encodeWithSelector(TempleLineOfCredit.ExceededCollateralAmonut.selector, alice, collateralAmount, withdrawalAmount));
+
+         vm.startPrank(alice);
+         tlc.withdrawCollateral(withdrawalAmount);
+         vm.stopPrank();
+    }
+
+    function testWithdrawWillUnderCollaterlizeLoan() external {
+        uint256 borrowAmount = uint(60_000e18);
+        uint256 collateralAmount = uint(100_000e18);
+        uint256 withdrawalAmount = uint(30_001e18);
+        _borrow(alice, collateralAmount, borrowAmount);
+         vm.expectRevert(abi.encodeWithSelector(TempleLineOfCredit.WillUnderCollaterlize.selector, alice, withdrawalAmount));
+
+         vm.startPrank(alice);
+         tlc.withdrawCollateral(withdrawalAmount);
+         vm.stopPrank();
+    }
+
+    function testWithdrawalSuccess() external {
+        uint256 borrowAmount = uint(60_000e18);
+        uint256 collateralAmount = uint(100_000e18);
+        uint256 withdrawalAmount = uint(10_000e18);
+        _borrow(alice, collateralAmount, borrowAmount);
+
+        uint256 collateralBalanceBefore = collateralToken.balanceOf(address(tlc));
+        uint256 aliceCollateralBalanceBefore = collateralToken.balanceOf(alice);
+
+         vm.startPrank(alice);
+         tlc.withdrawCollateral(withdrawalAmount);
+         (uint256 aliceCollateralAmount,,) = tlc.positions(alice);
+         vm.stopPrank();
+
+         assertEq(collateralBalanceBefore - withdrawalAmount, collateralToken.balanceOf(address(tlc)));
+         assertEq(aliceCollateralAmount, collateralAmount - withdrawalAmount);
+         assertEq(collateralToken.balanceOf(alice), aliceCollateralBalanceBefore + withdrawalAmount);
+    }
+
+
+    function testLiquidateSufficientCollateral() external {
+        uint256 borrowAmount = uint(70_000e18);
+        uint256 collateralAmount = uint(100_000e18);
+        _borrow(alice, collateralAmount, borrowAmount);
+        vm.expectRevert(abi.encodeWithSelector(TempleLineOfCredit.OverCollaterilized.selector, alice));
+
+        vm.prank(admin);
+        tlc.liquidate(alice);
+    }
+
+    function testLiquidateUnderWaterPositionSucessfully() external {
+        uint256 borrowAmount = uint(70_000e18);
+        uint256 collateralAmount = uint(100_000e18);
+        _borrow(alice, collateralAmount, borrowAmount);
+        vm.warp(block.timestamp + 1180 days);
+        uint256 totalDebt = tlc.getTotalDebtAmount(alice);
+        assertTrue(tlc.getCurrentCollaterilizationRatio(alice) < minCollateralizationRatio);
+        vm.prank(admin);
+        tlc.liquidate(alice);
```

Comment:
Should check events are emitted.

Also I don't think these edge cases are tested:
* `collateralSeized > positions[debtor].collateralAmount`
* `positions[account].debtAmount == 0`

Definitely worth you running forge coverage to see what's covered/what's not:
```
forge coverage --report lcov && genhtml lcov.info -o report --branch-coverage --legend
```